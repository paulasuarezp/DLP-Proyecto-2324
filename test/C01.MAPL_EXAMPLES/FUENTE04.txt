// Comparaciones e if

float menor, num;

read menor;

read num;
if (menor > num)
	menor = num

read num;
if (menor > num)
	menor = num

print menor;
