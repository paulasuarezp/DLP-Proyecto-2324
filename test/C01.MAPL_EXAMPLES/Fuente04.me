//*
Ejemplo Fuente04.
Comparaciones e if
*//

class Fuente04;
	global
		vars
			menor, num : DOUBLE;
			
	create
		main; 
			
	feature	main is
		/* local
			
		do
			read menor;
			read num;
			if menor > num then
				menor := num;
			end
			read num;
			if menor > num then
				menor := num;
			end
			print menor;	
	end /*main
	
end /* Fuente04

run main();