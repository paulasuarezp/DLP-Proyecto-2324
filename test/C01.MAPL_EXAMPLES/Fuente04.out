' Compilador mini Eiffel.
' Aquilino A. Juan.
' Versi√≥n 2024.2 (c)
#source "Fuente04.me"
#global menor: float
#global num: float
#func main
#ret void
CALL main
HALT

#line 14
main:

#line 18
PUSHA 0
INF
STOREF

#line 19
PUSHA 4
INF
STOREF

#line 20
PUSHA 0
LOADF
PUSHA 4
LOADF
GTF
JZ else_0000

#line 21
PUSHA 0
PUSHA 4
LOADF
STOREF
JMP endif_0000
else_0000:
endif_0000:

#line 23
PUSHA 4
INF
STOREF

#line 24
PUSHA 0
LOADF
PUSHA 4
LOADF
GTF
JZ else_0001

#line 25
PUSHA 0
PUSHA 4
LOADF
STOREF
JMP endif_0001
else_0001:
endif_0001:

#line 27
PUSHA 0
LOADF
OUTF

#line 27
RET 0, 0, 0
